### PR TITLE
chore(clerk-react,nextjs,shared): Allow react and react-dom 19.0.0-beta

### PR DESCRIPTION
**Pull request by nikosdouvlis**
With the next major release, NextJS@15 will deped on react and react-dom v19 that is still in beta. We are updating our peer dependencies for the related packages, otherwise the package managers will fail to resolve the package versions when users try to install or update the Clerk packages.

 Due to the semver specification for prerelease tags (https://semver.org/#spec-item-11), we can't be very exact here. Adding `19.0.0-beta` to the list will also allow `19.0.0-canary` and `19.0.0-rc` releases as well, but this is not something we can or need to solve at the moment.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/nextjs/package.json [hunk 1]
```diff
@@ -86,8 +86,8 @@
   },
   "peerDependencies": {
     "next": "^13.5.4 || ^14.0.3",
-    "react": ">=18",
-    "react-dom": ">=18"
+    "react": ">=18 || 19.0.0-beta",
```
Comment (BRKalow) [inline packages/nextjs/package.json:89]: given that `>=18` has no upper bound, is this change necessary? I'm guessing it might be because of the prerelease identifier.

In any case,  I think this should be:
```
>=18 || >=19.0.0-beta
```

Per https://semver.npmjs.com

Comment (nikosdouvlis) [reply packages/nextjs/package.json:89]: Oh damn it, I think I messed up here. 
It 100% needs to be `>=` or the check on the prerelease will always be false.
Let me test quick to make sure

**Review by BRKalow — APPROVED** [review]
